rep-2001.rst

```diff
@@ -38,26 +38,45 @@ We define three main entry points for ROS users.
  * desktop (recommended)
  * ros_base
  * ros_core
+ * ros_minimal
 
 Variants
 --------
 
+ROS Minimal
+'''''''''''
+
+The `ros_minimal` variant is a highly restricted set of ROS2 functionality.
+It targets resource- and tool-constrained environments where package managers may be limited or unavailable.
+The intention is that this variant can target cross-compiling for custom OS sysroots that may not have build tools or Python available that run on the target platform.
+It may not contain any GUI or Python dependencies.
+It may not have any dependencies outside a source workspace of its members with an exception of ubiquitous Linux libraries.
+Note that this variant explicitly does not include an implementation of either RMW or rcl_logging, users will need to choose their own that handle the dependencies that come with them.
+
+::
+
+ - ros_minimal:
+      packages: [ament_cmake, ament_cmake_ros, ament_index,
+		 class_loader, common_interfaces, console_bridge,
+                 libyaml_vendor, poco_vendor, rcl, rcl_interfaces,
+                 rcl_logging, rclcpp, rcpputils, rcutils, rmw,
+                 rmw_implementation, ros_tracing, rosidl, rosidl_defaults,
+                 rosidl_typesupport, spdlog_vendor, test_interface_files,
+                 tinydir_vendor, unique_identifier_msgs]
```

Comment:
historically despite the name "packages", this represent a list of repositories that one can clone and build as is (without the REP needing to explicitly list all packages in all the listed repos).
However some of these repositories contain python packages (e.g. `ament_index` repository includes `ament_index_cpp` and `ament_index_py` or `ament_cmake` includes many python packages). Should they be split out to keep this list "without any Python dependency"? or switch to listing explicitly packages and a recommended way to clone only those packages?

Reply:
That's a good point - I am definitely hoping to build a subset of packages within the involved repositories. I'll add a list of repos and the packages from them that will be built. Since the variants are described by a meta-package, though, if you build `--packages-up-to VARIANT` that's probably more correct than checking things out and building everything.